lib/utils.pm

```diff
@@ -998,22 +998,24 @@ sub run_scripted_command_slow {
 Returns tty number used designed to be used for root-console.
 When console is not yet initialized, we cannot get it from arguments.
 Since SLE 15 gdm is running on tty2, so we change behaviour for it and
-openSUSE distris.
+openSUSE distris, except for Xen PV (bsc#1086243).
 =cut
 sub get_root_console_tty {
-    return (sle_version_at_least('15') && !is_caasp) ? 6 : 2;
+    return (is_sle('15+') && !is_caasp && !check_var('VIRSH_VMM_TYPE', 'linux')) ? 6 : 2;
```

Comment:
Sorry for the confusing comment, but sle_version_at_least return trur for Tumbleweed and Leap, for example. So to be on safe side we cannot easily replace sle_version_at_least with is_sle here. But I guess it still means that check_var('VIRSH_VMM_TYPE', 'linux') check should be enough...

Reply:
Thanks for the explanation. I am still not sure what is the right `is_sle`/`is_leap`/`is_tumbleweed` replacement for `sle_version_at_least('15')`. But that's OK. Just adding the Xen PV check, which should not break anything, and leaving `sle_version_at_least` for someone who actually knows what needs to be done.

Reply:
Yeah, that's the legacy we have and the reason why we haven't simply replaced all sle_version_at_least usages. And it's almost impossible to understand where certain code it triggered, so usually we still have to run it and if no one complains, means it's fine to have it =)